packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/gateway/L1ERC20Gateway.sol

```diff
@@ -54,15 +54,16 @@ contract L1ERC20Gateway is L1ArbitrumExtendedGateway {
 
     // end of inline reentrancy guard
 
-    function outboundTransfer(
+    function outboundTransferCustomRefund(
```

Comment:
Same as the custom gateway, we want to override both the `outboundTransfer` and the `outboundTransferCustomRefund` to add in `nonReentrant`

Reply:
nice catch will add

Reply:
On second thought we don't need to have `nonReentrant` in `outboundTransfer` because I already marked it as non-virtual and `outboundTransfer` default to `outboundTransferCustomRefund` [here](https://github.com/OffchainLabs/arbitrum/blob/63cb2f38db044d92a81bafdf1b3a933af8d7a892/packages/arb-bridge-peripherals/contracts/tokenbridge/ethereum/gateway/L1ArbitrumGateway.sol#L201)

Reply:
yea true, nice